Добавлены страницы:

archibald7812 opened the pull request:
1. Страница окончания игры
2. Страница загрузки игры

### Какую задачу решаем
Верстка страниц

### Скриншоты/видяшка (если есть)

### TBD (если есть)

archibald7812 committed:
Добавлены страницы:

1. Страница окончания игры
2. Страница загрузки игры

IvanPeleshok reviewed (changes requested):
Необходим хотя бы один апрув от тимейта

vasilek-liza commented on packages/client/src/app/RootLayout.tsx:
```diff
@@ -44,6 +44,20 @@ const RootLayout = () => {
           </NavLink>
+          <NavLink
+            to={ERoutes.ENDGAME}
+            className={({ isActive }) =>
```
тут хорошо было бы использовать classnames
https://github.com/JedWatson/classnames

IvanPeleshok replied:
Предлагаю ознакомиться https://snyk.io/advisor/npm-package/classnames/example

vasilek-liza commented on packages/client/src/shared/images/Tank_1.png:
а почему ты берешь  png? мне кажется svg было бы здесь уместнее с точки зрения оптимизации)

archibald7812 replied:
Просто нашел картинки, обрезал и сохранил в png, можно и конвертировать в svg, так должно быть лучше, спасибо!

vasilek-liza commented on packages/client/src/pages/Main/index.tsx:
```diff
@@ -1,8 +1,8 @@
-import './main.module.css'
+//import styles from './gametostart.module.css'
```
если это в данный момент не используется, лучше удалить и не нести закомменченные строки в пр)

vasilek-liza replied:
то же самое на 5 строке

Ma1ve commented on packages/client/src/pages/GameToStart/index.tsx:
```diff
@@ -0,0 +1,17 @@
+        <H1>Loading...</H1>
+      </div>
+      <div className={styles.loader}>
+        <div className={styles.loaderIcon}></div>
```
Если данный Loader будет использоваться во всем проекте, можно создать отдельный компонент, который будет появляться при отправке данных на сервер и тд.

stepanpr replied:
Да, как вариант

IvanPeleshok replied:
Отличная идея
Унификация - ключ к простоте и порядку

Ma1ve commented on packages/client/src/pages/EndGame/index.tsx:
```diff
@@ -0,0 +1,19 @@
+    <div className={styles.endGameWrapper}>
+      <div className={styles.endGameText}>
+        <H1>Game over</H1>
+      </div>
```
В компонентах EndGame и GameToStart есть схожие стили, чтобы избежать дублирования, можно создать отдельный компонент на примере GameComponent  (если, конечно, EndGame и GameToStart не будут изменены в дальнейшем) : 
```
const GameComponent = ({ title, children }) => (
  <div className={commonStyles.wrapper}>
    <div className={commonStyles.text}>
      <H1>{title}</H1>
    </div>
    {children}
  </div>
);
```

stepanpr replied:
Да, так как функция схожа у компонента, то в идеале можно переделать потом чтобы был один компонент с двумя состояниями -  "начало игры" и "конец игры".

vasilek-liza commented on packages/client/src/pages/EndGame/endgame.module.css:
```diff
@@ -0,0 +1,17 @@
+  height: var(--content-height);
+}
+
+.endGameText > div {
```
лучше использовать классы

archibald7812 replied:
Это тег "<Н1>" я не стал его менять, но он не подразумевает пробрасывание внешних стилей в него. Если автор этого компонента не против, то можно добавить такую возможность.

Ma1ve reviewed (approved):
Хорошая работа!

vasilek-liza commented on packages/client/package.json:
```diff
@@ -32,8 +32,8 @@
   },
   "license": "MIT"
-}
+}
```
отсутствует пустая последняя строка в файле

stepanpr reviewed (approved):
Отличная работа!

archibald7812 committed:
Fix review issues

1. Добавлена последняя пустая строка в файлы
2. Немного изменена анимации лоадера

IvanPeleshok commented on packages/client/src/pages/GameToStart/gametostart.module.css:
```diff
@@ -0,0 +1,51 @@
+  height: var(--content-height);
+}
+
+.gameToStartText > div {
```
Ссылаться относительно дива не очень безопасно
Хорошим подходом будет пользоваться всегда классами, там практически всегда будет уверенность, что именно на этот элемент мы хотим повесить стили

IvanPeleshok commented on packages/client/src/pages/GameToStart/index.tsx:
```diff
@@ -0,0 +1,17 @@
+import H1 from '../../shared/ui/H1'
```
Какое практическое применение H1?
Семантически это див, никакой особой пользы он не несёт
Название не очень понятное, почему бы тогда не воспользоваться `<h1 /> `?

Предлагаю хотя бы дать название Title или что-то такое

IvanPeleshok reviewed (changes requested):
Проделана огромная работа!

Есть небольшие замечания у ваших коллег и у меня в частности
Просьба доработать :)

archibald7812 committed:
Fix review issues

1. Выделен сомпонент Loader
2. Добавлен компонент Title
3. Убраны стили для > div

IvanPeleshok reviewed (approved):
Проделана огромная работа
Замечательная реализация
Апрув
